### ISSUE TITLE
Receiver not registered error in cast demo app

### ISSUE THREAD

**Issue by volkansahin45** [google/ExoPlayer#4748]
### Issue description
Getting receiver not registered error after the video finished.

### Reproduction steps
Open cast test app.
Select tears of steel hls video and play on cast
seek to end of the video.
When the video finished you will see the error in logcat.

### Version of ExoPlayer being used
cloned the release-v2 github repo

### Device(s) and version(s) of Android being used
Samsung Galaxy S8 Plus Android 8.0.0

```
08-29 17:49:52.471 27916-27916/com.google.android.exoplayer2.castdemo E/MediaNotificationServiceImpl: Unregistering trampoline BroadcastReceiver failed
    java.lang.IllegalArgumentException: Receiver not registered: xc@69210e7
        at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1181)
        at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1456)
        at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:658)
        at xb.b(:com.google.android.gms.dynamite_dynamitemodulesc@12874023@12.8.74 (040400-204998136):111)
        at wt.a(:com.google.android.gms.dynamite_dynamitemodulesc@12874023@12.8.74 (040400-204998136):19)
        at nd.onTransact(:com.google.android.gms.dynamite_dynamitemodulesc@12874023@12.8.74 (040400-204998136):12)
        at android.os.Binder.transact(Binder.java:612)
        at com.google.android.gms.internal.cast.zza.transactAndReadExceptionReturnVoid(Unknown Source:7)
        at com.google.android.gms.cast.framework.media.zze.onDestroy(Unknown Source:5)
        at com.google.android.gms.cast.framework.media.MediaNotificationService.onDestroy(Unknown Source:2)
        at android.app.ActivityThread.handleStopService(ActivityThread.java:3707)
        at android.app.ActivityThread.-wrap27(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1806)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

**Comment by AquilesCanta** [comment]
Sorry for the delay on this! Can you please provide a bugreport, as requested in the issue template?

**Comment by volkansahin45** [comment]
I guess problem has solved with the 2.9.0 because I don't see the this error anymore. Thanks
